### PR TITLE
DOMA-3900 added feature toggling service integration

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**

### CODE REVIEW

File: apps/condo/next.config.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const defaultLocale = conf.DEFAULT_LOCALE
 const insuranceAppUrl = conf['INSURANCE_APP_URL']
 const JivoSiteWidgetId = conf['JIVO_SITE_WIDGET_ID']
 const TinyMceApiKey = conf['TINY_MCE_API_KEY']
+const featureToggleApiUrl = conf['FEATURE_TOGGLE_API_URL']
```
Comment (sitozzz) [inline apps/condo/next.config.js:26]: May be move to a single key with json content? Like `FEATURE_TOGGLE_CONFIG='{"url": ..., "apiKey":...}'`

File: apps/condo/domains/common/components/FeatureFlagsContext.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }, [updateContext, user])
+
+    useEffect(() => {
+        if (organization) {
+            updateContext({ organization: organization.id })
+        }
+    }, [organization, updateContext])
+
+    useEffect(() => {
+        fetch(FEATURES_ENDPOINT)
```
Comment (sitozzz) [inline apps/condo/domains/common/components/FeatureFlagsContext.tsx:55]: May be add check for existence for this and make a request only if url was successfully generated after parsing environment?

**Review by sitozzz — APPROVED** [review]
This is ok, but I think you need to test this could be working without `featureToggleConfig` provided

**Review by akarjakin — APPROVED** [review]

File: apps/condo/domains/common/components/FeatureFlagsContext.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    }, [updateContext, user])
+
+    useEffect(() => {
+        if (organization) {
+            updateContext({ organization: organization.id })
+        }
+    }, [organization, updateContext])
+
+    useEffect(() => {
```
Comment (pahaz) [inline apps/condo/domains/common/components/FeatureFlagsContext.tsx:54]: it looks like there is will not work with server side rendering

File: apps/condo/domains/common/components/FeatureFlagsContext.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import { useAuth } from '@condo/next/auth'
```
Comment (pahaz) [inline apps/condo/domains/common/components/FeatureFlagsContext.tsx:1]: probably it's better to create `featureflags` folder and move everything to that folder

File: apps/condo/pages/_app.tsx [hunk 5]
```diff
@@ -212,7 +213,9 @@ const MyApp = ({ Component, pageProps }) => {
                                         <LayoutContextProvider>
                                             <LayoutComponent menuData={<MenuItems/>} headerAction={HeaderAction}>
                                                 <RequiredAccess>
-                                                    <Component {...pageProps} />
+                                                    <FeatureFlagsProvider>
```
Comment (pahaz) [inline apps/condo/pages/_app.tsx:216]: Feature flag provider should be at the top. It should loaded before the layout

File: apps/condo/domains/common/components/FeatureFlagsContext.tsx (hunk 4 above)
Comment (pahaz) [reply apps/condo/domains/common/components/FeatureFlagsContext.tsx:1]: Could you please provide the `README` with examples in that folder

File: apps/condo/domains/common/components/FeatureFlagsContext.tsx (hunk 3 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/common/components/FeatureFlagsContext.tsx:54]: removed it

File: apps/condo/featureflags/featureToggleManager.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const { GrowthBook } = require('@growthbook/growthbook')
```
Comment (pahaz) [inline apps/condo/featureflags/featureToggleManager.js:1]: Is it a backend? Probably it’s a common / util

File: apps/condo/featureflags/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+***
+
+## Test it locally:
+1) [Raise the GrowthBook locally](https://docs.growthbook.io/self-host)
+2) Get API key from Settings -> API Keys in GrowthBook interface and pass it and API URL to `FEATURE_TOGGLE_CONFIG` in `.env`.
+
+   (Example: `FEATURE_TOGGLE_CONFIG='{"url": "http://localhost:3100/api/features", "apiKey": "key_prod_1234abcd"}'`)
+3) Create a new feature toggle in GrowthBook interface
+4) Use `featureToggleManager` for feature flags on the server and `FeatureFlagsContext` for feature flags on the client
+
```
Comment (pahaz) [inline apps/condo/featureflags/README.md:57]: Could you please add examples for json feature value.

**Review by pahaz — APPROVED** [review]

File: apps/condo/featureflags/featureToggleManager.js (hunk 6 above)
Comment (nomerdvadcatpyat) [reply apps/condo/featureflags/featureToggleManager.js]: moved featureflags folder to packages

File: packages/featureflags/FeatureFlagsContext.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import { GrowthBook, GrowthBookProvider, useGrowthBook } from '@growthbook/growthbook-react'
+import { get } from 'lodash'
+import getConfig from 'next/config'
+import { useRouter } from 'next/router'
+import React, { createContext, useCallback, useContext, useEffect } from 'react'
+
+const growthbook = new GrowthBook()
+
+interface IFeatureFlagsContext {
+    useFlag: (name: string) => boolean,
```
Comment (SavelevMatthew) [inline packages/featureflags/FeatureFlagsContext.tsx:10]: Hook as prop?

File: packages/featureflags/featureToggleManager.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+const { GrowthBook } = require('@growthbook/growthbook')
```
Comment (SavelevMatthew) [inline packages/featureflags/featureToggleManager.js:1]: Not specified in package.json, but used directly

File: packages/featureflags/package.json [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@condo/featureflags",
+  "version": "1.0.0",
+  "dependencies": {
+    "@condo/config": "workspace:^",
+    "@growthbook/growthbook-react": "^0.9.1"
+  },
+  "peerDependencies": {
+    "next": "^9.5.5",
+    "react": "^16.13.1"
```
Comment (SavelevMatthew) [inline packages/featureflags/package.json:10]: >=16
>=9

File: packages/featureflags/featureToggleManager.js (hunk 9 above)
Comment (nomerdvadcatpyat) [reply packages/featureflags/featureToggleManager.js:1]: done

File: packages/featureflags/package.json (hunk 10 above)
Comment (nomerdvadcatpyat) [reply packages/featureflags/package.json]: done

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**